### PR TITLE
Initial emacs support

### PR DESCRIPTION
**Pull request by jbouwman**
Basic emacs mode: fontification, indent, indexing of functions, and stubs for swank interaction.

This is initial work towards #1093 

Evaluation and error reporting will depend on the addition of a complete package form, and an additional compiler entry point.

### CODE REVIEW

File: emacs/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+add this directory to your load-path, and require the mode:
+
+    ;; Coalton
+    
+    (add-to-list 'load-path "~/git/coalton/emacs")
+    (require 'coalton-mode)
+
+## Usage
+
+There is an example .caolton file in test/types.coalton.
```
Comment (stylewarning) [inline emacs/README.md:21]: typo

File: emacs/coalton-mode.el [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+(require 'slime)
+(require 'slime-coalton)
+
+(add-to-list 'slime-contribs 'slime-coalton)
+
+(defconst coalton-mode-version
+  (eval-when-compile
+    (lm-version (or load-file-name buffer-file-name))))
+
+(defvar coalton-ts-repo
```
Comment (stylewarning) [inline emacs/coalton-mode.el:24]: can we put this in a self-contained place?

File: emacs/coalton-mode.el [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    table))
+
+(defvar coalton--debug nil
+  "Enable debugging.")
+
+

+;; Fontification
+
+(defconst coalton--builtins
```
Comment (stylewarning) [inline emacs/coalton-mode.el:61]: repr, declare, lisp

**Review by stylewarning — CHANGES_REQUESTED** [review]

**Comment by eliaslfox** [comment]
Could this go in another repo under coalton-lang?

**Comment by jbouwman** [comment]
> Could this go in another repo under coalton-lang?

Yes: could a maintainer create two new repositories?

1. coalton-mode
2. tree-sitter-coalton

**Comment by eliaslfox** [comment]
done

**Comment by jbouwman** [comment]
> done

thanks, however, these are empty and cannot be forked

tree-sitter-coalton can be directly initialized with the contents of https://github.com/jbouwman/tree-sitter-coalton

**Comment by stylewarning** [comment]
> > Could this go in another repo under coalton-lang?
> 
> 
> 
> Yes: could a maintainer create two new repositories?
> 
> 
> 
> 1. coalton-mode
> 
> 2. tree-sitter-coalton

@jbouwman you should have write access

File: emacs/README.md (hunk 1 above)
Comment (jbouwman) [reply emacs/README.md:21]: fixed

File: emacs/coalton-mode.el (hunk 2 above)
Comment (jbouwman) [reply emacs/coalton-mode.el:24]: this is now a URL for coalton-lang/tree-sitter-coalton

File: emacs/coalton-mode.el (hunk 3 above)
Comment (jbouwman) [reply emacs/coalton-mode.el:61]: done

**Comment by jbouwman** [comment]
This PR was converted into the initial commit to https://github.com/coalton-lang/coalton-mode

**Pull request closed without merging**